remotes/docker/resolver.go

```diff
@@ -286,6 +286,7 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 				if lastErr == nil {
 					lastErr = err
 				}
+				log.G(ctx).Debug(err.Error())
```

Comment:
Use `WithError` and have the message be something like "trying next host"

Reply:
kk and what do you think about log level... error, warn, debug, maybe info?

Reply:
```
INFO[2021-04-14T08:16:27.739730770-05:00] PullImage "gcr.io/hello-world"               
ERRO[2021-04-14T08:16:30.751322381-05:00] trying next host                              error="failed to do request: Head \"http://172.18.0.7:5000/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.7:5000: connect: no route to host" host="172.18.0.7:5000"
ERRO[2021-04-14T08:16:33.751089124-05:00] trying next host                              error="failed to do request: Head \"https://172.18.0.30:5010/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.30:5010: connect: no route to host" host="172.18.0.30:5010"
ERRO[2021-04-14T08:16:36.751018733-05:00] trying next host                              error="failed to do request: Head \"https://172.18.0.20:5007/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.20:5007: connect: no route to host" host="172.18.0.20:5007"
ERRO[2021-04-14T08:16:36.959594821-05:00] PullImage "gcr.io/hello-world" failed         error="failed to pull and unpack image \"gcr.io/hello-world:latest\": failed to resolve reference \"gcr.io/hello-world:latest\": unexpected status code [manifests latest]: 400 Bad Request"
```

Reply:
I feel like it should be Info

Reply:
kk  ...

Reply:
```
INFO[2021-04-14T13:21:24.889266889-05:00] PullImage "gcr.io/hello-world"               
INFO[2021-04-14T13:21:27.892926926-05:00] trying next host                              error="failed to do request: Head \"http://172.18.0.7:5000/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.7:5000: connect: no route to host" host="172.18.0.7:5000"
INFO[2021-04-14T13:21:30.894969373-05:00] trying next host                              error="failed to do request: Head \"https://172.18.0.30:5010/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.30:5010: connect: no route to host" host="172.18.0.30:5010"
INFO[2021-04-14T13:21:33.904204120-05:00] trying next host                              error="failed to do request: Head \"https://172.18.0.20:5007/v2/hello-world/manifests/latest?ns=gcr.io\": dial tcp 172.18.0.20:5007: connect: no route to host" host="172.18.0.20:5007"
ERRO[2021-04-14T13:21:34.112282331-05:00] PullImage "gcr.io/hello-world" failed         error="failed to pull and unpack image \"gcr.io/hello-world:latest\": failed to resolve reference \"gcr.io/hello-world:latest\": unexpected status code [manifests latest]: 400 Bad Request"
```
looks better... also added output for when a tag is not found on host to output the actual host used 
```
INFO[2021-04-14T13:41:56.909262733-05:00] PullImage "hello-world:foo"                  
INFO[2021-04-14T13:41:58.303650377-05:00] trying next host - response was http.StatusNotFound  host=registry-1.docker.io
ERRO[2021-04-14T13:41:58.304505701-05:00] PullImage "hello-world:foo" failed            error="rpc error: code = NotFound desc = failed to pull and unpack image \"docker.io/library/hello-world:foo\": failed to resolve reference \"docker.io/library/hello-world:foo\": docker.io/library/hello-world:foo: not found"
```